core/frontend/src/AccuDraw.ts

```diff
@@ -2305,7 +2303,7 @@ export class AccuDraw {
     if (!this._distanceRoundOff.active || !this._distanceRoundOff.units.size)
       return undefined;
 
-    let roundValue = this._distanceRoundOff.units.values().next().value!; // eslint-disable-line @typescript-eslint/no-non-null-assertion
+    let roundValue = this._distanceRoundOff.units.values().next().value!;
```

Comment:
Are we no longer flagging non-null assertions as? If so why?

Reply:
The rule is enabled for non-test files in core-backend.  As far as I can tell it's never been enabled in core-frontend.

The comment here was just copypasta I accidentally inserted in #7232

Reply:
That is dismaying. Do you know why it's not enabled in the base config? I have lost count of how many bugs/crashes in production have been traced back to somebody's overly-optimistic exclamation mark.

Reply:
If that somebody is willing to add `!` to make life easier, what makes you think she/he won't add `eslint-disable-...` for it too? TS already lets you know that there's a problem and allows you to ignore it by adding the exclamation mark. I don't think adding another warning and another way to ignore it will solve anything - it's up to the developer and reviewer.

Reply:
Exclamation mark is way too easy to overlook during review, vs `assert`.
[Exhibit A](https://github.com/iTwin/itwinjs-backlog/issues/1243#issuecomment-2333862507).
```ts
  private ecefToPixelFraction(point: Point3d, applyTerrain: boolean): Point3d {
    const cartoGraphic = Cartographic.fromEcef(point)!;
    return Point3d.create(this.longitudeToXFraction(cartoGraphic.longitude), this.latitudeToYFraction(cartoGraphic.latitude), applyTerrain ? cartoGraphic.height : 0);
```

But this PR isn't about changing the lint config.